rust-runtime/smithy-eventstream/src/str_bytes.rs

```diff
@@ -106,34 +61,43 @@ impl From<&'static str> for StrBytes {
     fn from(value: &'static str) -> Self {
         StrBytes {
             bytes: Bytes::from(value),
-            valid_utf8: AtomicU8::new(State::Valid.as_u8()),
         }
     }
 }
 
-impl From<&'static [u8]> for StrBytes {
-    fn from(value: &'static [u8]) -> Self {
-        StrBytes {
-            bytes: Bytes::from(value),
-            valid_utf8: AtomicU8::new(State::Unknown.as_u8()),
+impl TryFrom<&'static [u8]> for StrBytes {
```

Comment:
usually I've seen this as an explicit `from_static` inherent method because if we ever want to add a `TryFrom<&'a [u8]>` these would conflict

Reply:
I'm following `bytes`'s lead here. Added `try_copy_from_slice` and `copy_from_str` functions for the non-static use-case.